### PR TITLE
ROX-12012: Additional config for auth provider

### PR DESCRIPTION
**Pull request by dhaus67**
## Description

Set the `disable_offline_access_scope` config as well as the required attributes to match the owner's org_id.

This will ensure only clients within the same organization of the owner can login to central using the created auth provider.

## Test manual

```
# Start fleet manager
./fleet-manager serve --dataplane-cluster-config-file=<your-config> --rhsso-client-secret-file=<client secret>
# Start fleetshard synchronizer
CREATE_AUTH_PROVIDER=true CLUSTER_ID=1234567890abcdef1234567890abcdef OCM_TOKEN=$(ocm token --refresh) ./fleetshard-sync
# Start operator
cdrox && make -C operator install run
# Login to staging OCM with offline token of dev-acs-ms-corporate-test
ocm login --url staging --token-url https://sso.stage.redhat.com/auth/realms/redhat-external/protocol/openid-connect/token --token=<offline token>
# Create managed central
./scripts/create-central.sh
# Make sure reconciliation of managed central works as expected, i.e. no error logs and auth provider creation is successful 
```
Access the UI of central and the auth provider list, observe within the API response the correct settings for the auth provider:
Screenshot 2022-07-29 at 21 02 18

Login with the user `dev-acs-ms-corporate-test`, ensure login works successfully.

Visit [offline tokens](https://sso.stage.redhat.com/auth/realms/redhat-external/account/applications) with `dev-acs-ms-corporate-test` user and verify no offline token has been created for application `rhacs-ms-dev`:
Screenshot 2022-07-31 at 21 07 37

### CODE REVIEW

**Comment by dhaus67** [comment]
Current dependencies on/for this PR:
* main
  * **PR #246** <a href="https://app.graphite.dev/github/pr/stackrox/acs-fleet-manager/246" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/stackrox/acs-fleet-manager/246?utm_source=stack-comment).

File: fleetshard/pkg/central/reconciler/init_auth.go [hunk 1]
```diff
@@ -91,13 +91,20 @@ func createAuthProviderRequest(central private.ManagedCentral) *storage.AuthProv
...
 			// TODO: ROX-11619: make configurable
-			"issuer":        "https://sso.stage.redhat.com/auth/realms/redhat-external",
-			"client_id":     central.Spec.Auth.ClientId,
-			"client_secret": central.Spec.Auth.ClientSecret,
-			"mode":          "post",
+			"issuer":                 "https://sso.stage.redhat.com/auth/realms/redhat-external",
+			"client_id":              central.Spec.Auth.ClientId,
+			"client_secret":          central.Spec.Auth.ClientSecret,
+			"mode":                   "post",
+			"disable_offline_access": "true",
```
Comment (ivan-degtiarenko) [inline fleetshard/pkg/central/reconciler/init_auth.go:98]: Sorry, it looks like the ticket is misleading - the actual name of the field is `disable_offline_access_scope`:
https://github.com/stackrox/stackrox/blob/a539c76f0f3d50a42ad2e5bafc6b6e3b3e500e01/pkg/auth/authproviders/oidc/backend_impl.go#L37-L37

Would it be possible to ask you to fix it and verify that it actually works on the last released version of RHACS?
For that you need to:
* log in using sso.stage.redhat.com account
* Check that no offline tokens were created here: https://sso.stage.redhat.com/auth/realms/redhat-external/account/applications

**Review by ivan-degtiarenko — CHANGES_REQUESTED** [review]
Also, could you test that org_id restriction works within created Central?

For that you would need to use the same identity to:
* send "create central" request to fleet-manager
* log in to central

File: fleetshard/pkg/central/reconciler/init_auth.go (hunk 1 above)
Comment (dhaus67) [reply fleetshard/pkg/central/reconciler/init_auth.go:98]: I could have checked it as well 🥴 

Fixed the setting + verified that no offline token is created (adjusted the manual test steps in the description accordingly).

**Review by ivan-degtiarenko — APPROVED** [review]

**Pull request merged**